go-controller/pkg/ovn/ovn.go

```diff
@@ -542,15 +543,29 @@ func (oc *Controller) iterateRetryPods(updateAll bool) {
 		backoff := (podEntry.backoffSec * time.Second) + (time.Duration(rand.Intn(500)) * time.Millisecond)
 		podTimer := podEntry.timeStamp.Add(backoff)
 		if updateAll || now.After(podTimer) {
-			klog.Infof("%s retry pod setup", podDesc)
+			if !podEntry.ignoreAddRetry {
+				klog.Infof("%s retry pod setup", podDesc)
 
-			if oc.ensurePod(nil, kPod, true) {
-				klog.Infof("%s pod setup successful", podDesc)
-				delete(oc.retryPods, uid)
-			} else {
-				klog.Infof("%s setup retry failed; will try again later", podDesc)
-				oc.retryPods[uid] = &retryEntry{pod, time.Now(), podEntry.backoffSec, false}
+				if oc.ensurePod(nil, kPod, true) {
+					klog.Infof("%s pod setup successful", podDesc)
```

Comment:
Also, the lines earlier in the function like line 527:
```
kPod, err := oc.watchFactory.GetPod(pod.Namespace, pod.Name)
		if err != nil && errors.IsNotFound(err) {
			klog.Infof("%s pod not found in the informers cache, not going to retry pod setup", podDesc)
			delete(oc.retryPods, uid)
			continue
		}
```
Will need to be modified. Post delete the pod wont be in the informers cache, and we don't want to get the latest pod anyway, we want to delete things based on the state of the pod when we knew about it.